src/realm/object-store/c_api/schema.cpp

```diff
@@ -54,6 +54,14 @@ RLM_API bool realm_schema_validate(const realm_schema_t* schema)
     });
 }
 
+RLM_API bool realm_set_schema(realm_t* realm, const realm_schema_t* schema)
```

Comment:
Maybe `realm_update_schema()` would actually be a better name, since the behavior is controlled by the SchemaMode?